src/Shared/Recruit.Vacancies.Client/Application/Validation/Fluent/FluentVacancyValidator.cs

```diff
@@ -0,0 +1,297 @@
+                .WithRuleId(VacancyRuleSet.StartDate);
+        }
+
+        private void ValidateTrainingProgramme()
+        {
+            RuleFor(x => x.Programme.Id)
+                .NotEmpty()
+                    .WithMessage("Select  apprenticeship training")
+                    .WithErrorCode("25")
+                .WithRuleId(VacancyRuleSet.TrainingProgramme)
+                .RunCondition(VacancyRuleSet.TrainingProgramme);
+        }
+
+        private void ValidateDuration()
+        {
+            RuleFor(x => x.Wage.DurationUnit)
+                .NotEmpty()
+                    .WithMessage("Enter the expected duaration")
+                    .WithErrorCode("34")
+                .IsInEnum()
+                    .WithMessage("Enter the expected duaration")
+                    .WithErrorCode("34")
+                .RunCondition(VacancyRuleSet.Duration)
+                .WithRuleId(VacancyRuleSet.Duration);
+
+            RuleFor(x => x.Wage.Duration)
+                .Cascade(CascadeMode.StopOnFirstFailure)
+                .NotEmpty()
+                    .WithMessage("Enter the expected duaration")
+                    .WithErrorCode("34")
+                .GreaterThan(0)
+                    .WithMessage("Enter the expected duaration")
+                    .WithErrorCode("34")
+                .Must((vacancy, value) => 
+                {
+                    if (vacancy.Wage.DurationUnit == DurationUnit.Month && value < 12)
+                    {
+                        return false;
+                    }
+
+                    return true;
+                })
+                    .WithMessage("The expected duration must be at least 12 months (52 weeks)")
+                    .WithErrorCode("36")
+                .RunCondition(VacancyRuleSet.Duration)
+                .WithRuleId(VacancyRuleSet.Duration);
+        }
+
+        private void ValidateWorkingWeek()
+        {
+            RuleFor(x => x.Wage.WorkingWeekDescription)
+                .Cascade(CascadeMode.StopOnFirstFailure)
+                .NotEmpty()
+                    .WithMessage("Enter the working week")
+                    .WithErrorCode("37")
+                .ValidFreeTextCharacters()
+                    .WithMessage("The working week contains some invalid characters")
+                    .WithErrorCode("38")
+                .MaximumLength(250)
+                    .WithMessage("The working week must not be more than {MaxLength} characters")
+                    .WithErrorCode("39")
+                .RunCondition(VacancyRuleSet.WorkingWeekDescription)
+                .WithRuleId(VacancyRuleSet.WorkingWeekDescription);
+        }
+
+        private void ValidateWeeklyHours()
+        {
+            RuleFor(x => x.Wage.WeeklyHours)
+                .NotEmpty()
+                    .WithMessage("Enter the hours per week")
+                    .WithErrorCode("40")
+                .GreaterThan(16)
```

Comment:
I think 16 is the minimum. So `GreaterThan(15)` and `LesThan(49)` or equivalent

Reply:
The Error messages on Confluence didn't match the story. Messages in Confluence have been updated and code brought into line.